common/cosmwasm-smart-contracts/vesting-contract/src/lib.rs

```diff
@@ -42,6 +44,48 @@ impl PledgeData {
     }
 }
 
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub enum PledgeCap {
+    Percent(u8),
+    Absolute(Uint128),
+}
+
+impl FromStr for PledgeCap {
+    type Err = String;
+
+    fn from_str(cap: &str) -> Result<Self, Self::Err> {
+        match cap.parse::<f32>() {
+            Ok(f) => {
+                if f > 0. && f <= 1. {
+                    Ok(PledgeCap::Percent((f * 100.) as u8))
+                } else {
+                    Err(format!("Floats have to be > 0. and <= 1., got {}", f))
+                }
+            }
+            Err(_) => match cap.parse::<u128>() {
+                Ok(i) => Ok(PledgeCap::Absolute(Uint128::from(i))),
+                Err(_e) => Err(format!("Could not parse {} as f32 or u128", cap)),
+            },
+        }
+    }
+}
+
+impl Default for PledgeCap {
+    fn default() -> Self {
+        PledgeCap::Absolute(Uint128::from(100_000_000_000u128))
+    }
+}
+
+impl PledgeCap {
+    pub fn valid(&self) -> bool {
+        match &self {
+            PledgeCap::Percent(p) => *p > 0 && *p <= 100,
```

Comment:
If you're locking it between 0-100%, we already have a `Percent` type with that constraint (https://github.com/nymtech/nym/blob/develop/common/cosmwasm-smart-contracts/mixnet-contract/src/types.rs#L31) though it'd probably have to be moved outside mixnet-contract specific types then